client/my-sites/site-settings/publishing-tools/post-by-email.tsx

```diff
@@ -0,0 +1,163 @@
+
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const handleSwitchForSimpleSite = ( checked: boolean ) => {
+		if ( ! selectedSiteId ) {
+			return;
+		}
+
+		simpleSiteSwitchPostByEmail( checked, {
+			onSuccess: () => {
+				dispatch( successNotice( translate( 'Settings saved successfully!' ), noticeConfig ) );
+			},
+			onError: () => {
+				dispatch(
+					errorNotice( translate( 'There was a problem saving your changes.' ), noticeConfig )
+				);
+			},
+		} );
+	};
+
+	const handleRegenerate = () => {
+		if ( ! selectedSiteId ) {
+			return;
+		}
+
+		if ( siteIsJetpack ) {
+			dispatch( jetpackRegeneratePostByEmail( selectedSiteId ) );
+			return;
+		}
+
+		simpleSiteRegeneratePostByEmail( undefined, {
+			onSuccess: () => {
+				dispatch( successNotice( translate( 'Address regenerated successfully!' ), noticeConfig ) );
+			},
+			onError: () => {
+				dispatch(
+					errorNotice( translate( 'There was a problem regenerating the address.' ), noticeConfig )
+				);
+			},
+		} );
+	};
+
+	const email = address && address !== 'regenerate' ? address : '';
+	const isActive = siteIsJetpack
+		? jetpackPostByEmailIsActive
+		: simpleSitePostByEmailSettings?.isEnabled;
+	const labelClassName =
+		moduleUnavailable || jetpackRegeneratingPostByEmail || ! isActive ? 'is-disabled' : undefined;
+
+	return (
+		<>
+			<FormFieldset>
+				<SupportInfo
+					text={ translate(
+						'Allows you to publish new posts by sending an email to a special address.'
+					) }
+					link={
+						siteIsAtomic
+							? localizeUrl( 'https://wordpress.com/support/post-by-email/' )
+							: 'https://jetpack.com/support/post-by-email/'
+					}
+					privacyLink={ ! siteIsAtomic }
+				/>
+				{ siteIsJetpack ? (
+					<JetpackModuleToggle
+						siteId={ selectedSiteId }
+						moduleSlug="post-by-email"
+						label={ translate( 'Publish posts by sending an email' ) }
+						disabled={ isFormPending || moduleUnavailable }
+					/>
+				) : (
+					<ToggleControl
+						checked={ !! simpleSitePostByEmailSettings?.isEnabled }
+						disabled={ isSimpleSitePendingSwitch || isSimpleSitePendingRegenerate }
+						label={ translate( 'Post by Email' ) }
+						onChange={ handleSwitchForSimpleSite }
+					/>
+				) }
+
+				<div className="publishing-tools__module-settings site-settings__child-settings">
+					<FormLabel className={ labelClassName }>
+						{ translate( 'Send your new posts to this email address:' ) }
+					</FormLabel>
+					<ClipboardButtonInput
+						className="publishing-tools__email-address"
+						disabled={ jetpackRegeneratingPostByEmail || ! isActive || moduleUnavailable }
+						value={ siteIsJetpack ? email : simpleSitePostByEmailSettings?.email }
+					/>
+					<Button
+						onClick={ handleRegenerate }
+						disabled={
+							isFormPending ||
+							jetpackRegeneratingPostByEmail ||
+							isSimpleSitePendingRegenerate ||
+							! isActive ||
+							!! moduleUnavailable
```

Comment:
We have different condition for disabled state here and here (https://github.com/Automattic/wp-calypso/pull/90819/files#diff-1a2d2a711ac2389148e7764b72177d565e3e6a88fea709f208906a031a5a64c5R142)

1) Let's make it the same (I thing we can add `isFormPending` to the "Copy" button)
2) Let's create variable for both cases to avoid duplication

Reply:
This one was actually a bit of an interesting rabbit hole once I started looking at it more closely.

There are five `disabled` props in play:
1. Jetpack toggle
2. Simple site toggle
3. Copy button,
4. Regenerate button
5. `is-disabled` classname on the `FormLabel`. 
6. 
7. Across those five items, we actually had _four_ different conditions. Technically only four ever appear at once, but still... that's way too many combinations of conditions!

To clean this up, I've made it so the three items that apply to both flows are the same using a new, single variable to check the relevant conditions. I've also made it so `isFormPending` is included in all conditions, as that's not flow-specific and checks to see if any settings are being requested for the current site.

One other interesting discovery... for the new Simple site toggle, we disable the toggle itself if the setting is being toggled or if the email is being regenerated. I think that's something I initially included when taking inspiration from earlier versions of your Post by Voice PR.

We don't actually have the same behavior for Jetpack, because:
1. We don't have a Jetpack version of `isSimpleSitePendingToggle` (the `isUpdatingJetpackSettings` selector might suffice, but I'll need to look at it a bit more)
2. While we do have `jetPackRegeneratingPostByEmail`, I don't see it actually _doing_ anything. It doesn't apply a disabled state, or change the button text the way the code currently asks it to. It doesn't even log out as `true` when the regeneration button is pressed. I'm not sure if it just isn't implemented correctly, or if there's some other problem. 

When I'm back on tuesday, I'll prioritize looking into `jetPackRegeneratingPostByEmail` and `isUpdatingJetpackSettings` to see if they can serve the same roles as our Simple counterparts, or if they should just be excluded entirely for now.

But either way, the disabled logic should be a little simpler now... and might become more so if we decide to go without `jetPackRegeneratingPostByEmail`.